src/sentry/api/endpoints/artifact_lookup.py

```diff
@@ -235,140 +203,35 @@ def try_resolve_urls(
     except Exception as exc:
         logger.error("Failed to read", exc_info=exc)
 
-    # TODO: should we rather return an error from the API in case the release is not found?
-    if release is None:
-        return list()
-
-    remaining_urls = collect_legacy_artifact_bundles_containing_urls(
-        remaining_urls, release, dist, bundle_file_ids
-    )
-    if not remaining_urls:
-        return list()
-
-    # And last but not least, we want to look up legacy individual release files
-    return get_releasefiles_matching_urls(remaining_urls, release)
+    return release, dist
 
 
-def collect_release_artifact_bundles_containing_urls(
-    urls: List[str],
-    project: Project,
-    release_name: str,
-    dist_name: Optional[str],
-    bundle_file_ids: Set[int],
-) -> List[str]:
-    releases_with_bundles = (
-        ReleaseArtifactBundle.objects.filter(
-            organization_id=project.organization.id,
-            release_name=release_name,
-            # In case no dist is provided, we will fall back to "" which is the NULL equivalent for our tables.
-            # See `_create_artifact_bundle` in `src/sentry/tasks/assemble.py` for the reference.
-            dist_name=dist_name or "",
+def get_legacy_release_bundles(release: Release, dist: Optional[Distribution]):
+    return set(
+        ReleaseFile.objects.select_related("file")
+        .filter(
+            release_id=release.id,
+            dist_id=dist.id if dist else None,
+            artifact_count=0,
```

Comment:
I would put a comment here, explaining the rationale for such query. From the outside's perspective it might be a bit difficult to infer.

Reply:
https://github.com/getsentry/sentry/blob/ef8e70e3996daa576eac291194daeb2fa77b1865/src/sentry/models/releasefile.py#L386-L393 here is the link to the line responsible for that behavior.